lib/source/Instruction.cpp

```diff
@@ -1614,6 +1633,42 @@ spvgentwo::Instruction* spvgentwo::Instruction::toSpecOp()
 	return this;
 }
 
+spvgentwo::Instruction* spvgentwo::Instruction::opPtrEqual(Instruction* _pLeftPtr,
+	Instruction* _pRightPtr)
+{
+	auto module = getModule();
+
+	if (_pLeftPtr->getType()->isPointer() && _pRightPtr->getType()->isPointer()
```

Comment:
_pLeftPtr->getType() is not exactly a cheap operation, it might make sense to query the const Type* left = _pLeftPtr->getType(); before the if statement so you cont have to call getType twice